src/transformStateWithClones.js

```diff
@@ -1,13 +1,39 @@
-'use strict';
+const ACTION_TYPE_CLEAR = 'clear';
+const ACTION_TYPE_ADD_PROPERTIES = 'addProperties';
+const ACTION_TYPE_REMOVE_PROPERTIES = 'removeProperties';
 
 /**
- * @param {Object} state
- * @param {Object[]} actions
+ * Оновлює стан об'єкта на основі масиву дій і повертає історію змін.
  *
- * @return {Object[]}
+ * @param {Object} initialState - Початковий стан об'єкта.
+ * @param {Object[]} actions - Масив дій, які будуть застосовані до стану.
+ * @returns {Object[]} - Масив станів після кожної дії.
  */
-function transformStateWithClones(state, actions) {
-  // write code here
+function transformStateWithClones(initialState, actions) {
+  const stateHistory = [];
+  let currentState = { ...initialState };
```

Comment:
The name `currentState` for the object copy is acceptable, but it's recommended to use a more descriptive name such as `stateCopy` to clarify that it is a copy of the state at a certain point in time, following the [NAMING] guidelines.